src/JPEGView/ImageLoadThread.cpp

```diff
@@ -641,6 +669,75 @@ void CImageLoadThread::ProcessReadWEBPRequest(CRequest * request) {
 	}
 }
 
+void CImageLoadThread::ProcessReadJXLRequest(CRequest* request) {
+	bool bUseCachedDecoder = false;
+	const wchar_t* sFileName;
+	sFileName = (const wchar_t*)request->FileName;
+	if (sFileName != m_sLastJxlFileName) {
+		DeleteCachedJxlDecoder();
+	} else {
+		bUseCachedDecoder = true;
+	}
+
+	HANDLE hFile;
+	if (!bUseCachedDecoder) {
+		hFile = ::CreateFile(request->FileName, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, 0, NULL);
+		if (hFile == INVALID_HANDLE_VALUE) {
+			return;
+		}
+	}
+	char* pBuffer = NULL;
+	UINT nPrevErrorMode = SetErrorMode(SEM_FAILCRITICALERRORS);
+	try {
+		unsigned int nFileSize;
+		unsigned int nNumBytesRead;
+		if (!bUseCachedDecoder) {
+			// Don't read too huge files
+			nFileSize = ::GetFileSize(hFile, NULL);
+			if (nFileSize > MAX_JXL_FILE_SIZE) {
+				request->OutOfMemory = true;
+				::CloseHandle(hFile);
+				return;
+			}
+
+			pBuffer = new(std::nothrow) char[nFileSize];
+			if (pBuffer == NULL) {
+				request->OutOfMemory = true;
+				::CloseHandle(hFile);
+				return;
+			}
+		} else {
+			nFileSize = 0; // to avoid compiler warnings, not used

```

Comment:
is this uninitialized variable warning?  does setting it up when declaring it (line 692) remove the warning?

Reply:
Yes to both. If you don't initialize it and run with the debugger, it will throw `Run-Time Check Failure #3 - The variable 'nFileSize' is being used without being initialized.` when it gets passed to ReadImage, even though `nFileSize` won't actually be used anywhere.